File: 2_Kernel/JavaKernel/src/main/java/com/seagosoft/tws/data/TickByTickAllLastData.java
package com.seagosoft.tws.data;

import com.ib.client.Decimal;
import com.ib.client.TickAttribLast;

public class TickByTickAllLastData {

    public int reqId;
    public int tickType;
    public long time;
    public double price;
    public Decimal size;
    public TickAttribLast tickAttribLast;
    public String exchange;
    public String specialConditions;

    public TickByTickAllLastData(int reqId, int tickType, long time,
                                 double price, Decimal size, TickAttribLast tickAttribLast,
                                 String exchange, String specialConditions) {
        this.reqId = reqId;
        this.tickType = tickType;
        this.time = time;
        this.price = price;
        this.size = size;
        this.tickAttribLast = tickAttribLast;
        this.exchange = exchange;
        this.specialConditions = specialConditions;
    }
}

File: 3_OrderSystem/KotlinEWrapperBackend/src/main/kotlin/com/seagosoft/tws/wrapper/structs/TickByTickAllLastData.kt
package com.seagosoft.tws.wrapper.structs

import com.ib.client.Decimal
import com.ib.client.TickAttribLast


class TickByTickAllLastData(
    var reqId: Int, var tickType: Int, var time: Long,
    var price: Double, var size: Decimal, var tickAttribLast: TickAttribLast,
    var exchange: String, var specialConditions: String
)